splight_lib/client/datalake/builder.py

```diff
@@ -3,22 +3,30 @@
 from splight_lib.client.datalake.abstract import AbstractDatalakeClient
 from splight_lib.client.datalake.local_client import LocalDatalakeClient
 from splight_lib.client.datalake.remote_client import (
-    BufferedRemoteDatalakeClient,
-    RemoteDatalakeClient,
+    BufferedAsyncRemoteDatalakeClient,
+    BufferedSyncRemoteDataClient,
+    SyncRemoteDatalakeClient,
 )
+from splight_lib.constants import DL_CLIENT_ALLOWED_TYPES
+
+dl_client_type_map = {
+    "buffered_async": BufferedAsyncRemoteDatalakeClient,
+    "buffered_sync": BufferedSyncRemoteDataClient,
+    "sync": SyncRemoteDatalakeClient,
+    "local": LocalDatalakeClient,
+}
```

Comment:
```suggestion
DL_CLIENT_TYPE_MAP = {
    "buffered_async": BufferedAsyncRemoteDatalakeClient,
    "buffered_sync": BufferedSyncRemoteDataClient,
    "sync": SyncRemoteDatalakeClient,
    "local": LocalDatalakeClient,
}
```

Also, I would use the Enum here.